core/src/main/java/io/grpc/StreamTracer.java

```diff
@@ -62,32 +62,45 @@ public void inboundMessage() {
   }
 
   /**
+   * An outbound message has been completely serialized, revealing its wire size.
+   */
+  public void outboundMessageSerialized(long wireSize) {
```

Comment:
I think other languages will have the same problem getting the message wire size with stream-based compression, because message boundaries have diminish in the compressed stream. Uncompressed size will be more feasible.